code/lib/core-server/src/utils/StoryIndexGenerator.ts

```diff
@@ -312,7 +314,9 @@ export class StoryIndexGenerator {
     const hasAutodocsTag = entries.some((entry) => entry.tags.includes(AUTODOCS_TAG));
     const isStoriesMdx = entries.some((entry) => entry.tags.includes(STORIES_MDX_TAG));
     const createDocEntry =
-      autodocs === true || (autodocs === 'tag' && hasAutodocsTag) || isStoriesMdx;
+      (autodocs === true && !entries.some((entry) => entry.tags.includes(NO_AUTODOCS_TAG))) ||
```

Comment:
Makes sense to pull this out to `hasNoAutodocsTag`, and then `hasAutodocsTag = !hasNoAutodocs & ...`, or even to call `combineTags` on the tags first.